File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/bluetooth/AlertMapper.java
package com.lattis.ellipse.bluetooth;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.network.base.AbstractDataMapper;
import com.lattis.ellipse.domain.model.Alert;

import javax.inject.Inject;

public class AlertMapper extends AbstractDataMapper<Alert, io.lattis.ellipse.sdk.model.Alert> {

    @Inject
    public AlertMapper() {
    }

    @NonNull
    @Override
    public io.lattis.ellipse.sdk.model.Alert mapIn(@NonNull Alert alert) {

         if(alert == null)
             return io.lattis.ellipse.sdk.model.Alert.OFF;

        switch (alert){
            case OFF:
                return io.lattis.ellipse.sdk.model.Alert.OFF;
            case THEFT:
                return io.lattis.ellipse.sdk.model.Alert.THEFT;
            case CRASH:
                return io.lattis.ellipse.sdk.model.Alert.CRASH;
            default:return null;
        }
    }

    @NonNull
    @Override
    public Alert mapOut(@NonNull io.lattis.ellipse.sdk.model.Alert alert) {

        if(alert == null)
            return Alert.OFF;

        switch (alert){
            case OFF:
                return Alert.OFF;
            case THEFT:
                return Alert.THEFT;
            case CRASH:
                return Alert.CRASH;
            default:return null;
        }
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/mapper/bluetooth/AlertMapper.kt
package com.lattis.data.mapper.bluetooth

import com.lattis.data.mapper.AbstractDataMapper
import com.lattis.domain.models.Alert
import javax.inject.Inject

class AlertMapper @Inject constructor() :
    AbstractDataMapper<Alert, io.lattis.ellipse.sdk.model.Alert>() {
    override fun mapIn(alert: Alert?): io.lattis.ellipse.sdk.model.Alert {
        return if (alert == null) io.lattis.ellipse.sdk.model.Alert.OFF else when (alert) {
            Alert.OFF -> io.lattis.ellipse.sdk.model.Alert.OFF
            Alert.THEFT -> io.lattis.ellipse.sdk.model.Alert.THEFT
            Alert.CRASH -> io.lattis.ellipse.sdk.model.Alert.CRASH
            else -> io.lattis.ellipse.sdk.model.Alert.OFF
        }
    }

    override fun mapOut(alert: io.lattis.ellipse.sdk.model.Alert?): Alert {
        return if (alert == null) Alert.OFF else when (alert) {
            io.lattis.ellipse.sdk.model.Alert.OFF -> Alert.OFF
            io.lattis.ellipse.sdk.model.Alert.THEFT -> Alert.THEFT
            io.lattis.ellipse.sdk.model.Alert.CRASH -> Alert.CRASH
            else -> Alert.OFF
        }
    }
}